demo/imgdetect/js/main.js

```diff
@@ -1,15 +1,47 @@
 $(document).ready(function() {
 
+  $('#service_select').select2();
+
+  var listServices = function() {
+
+    var service_url = "/api/info";
+
+    $.ajax({
+      type: "GET",
+      url: service_url,
+      dataType: 'json',
+      success: function(data) {
+
+        if(data.head.services && data.head.services.length > 0) {
+          $('#serviceForm').removeClass('bg-info')
+            .find('span').html('Select a service: ');
+
+          $('#service_select').select2({
+            data: data.head.services.map(function(item) {
+              return {id: item.name, text: item.name};
+            })
+          });
+          $('#service_select').show();
+          $('#uploadForm').removeClass('hidden');
+        } else {
+          $('#serviceForm').removeClass('bg-info')
+            .addClass('bg-danger')
+            .html('No service found, please set a service on DeepDetect server.');
+        }
+      }
+    });
+  };
+
   $('a#urlSubmit').click(function() {
 
     $('.loading').removeClass('hidden');
 
     var post_url = "/api/predict";
     var post_data = {
-      service: "imageserv",
+      service: $('#service_select').val(),
       parameters: {
         input: {width: 224, height: 224},
```

Comment:
This line should be removed or commented out as 224x224 is model specific. Any call with this on a model with inputs different than 224x224 yields a bad param error, which is normal (tested btw).